### PR TITLE
docs: fix incorrect demonstration

### PR DESCRIPTION
**Pull request by sm-stack**
## Summary by CodeRabbit

- **Documentation**
	- Updated the challenge process details to remove ambiguity about the handling of duplicate lines and invalid outputs.
	- Clarified the validator's actions regarding incorrect output roots in the derivation documentation.
	- Enhanced the glossary with a definition and explanation of on-chain interactive proofs.

### CODE REVIEW

File: specs/challenge.md [hunk 1]
```diff
@@ -48,8 +48,6 @@ In the ZK fault-proof challenge process, the following undeniable bug might aris
 
 - The deletion of a valid output due to two valid and contradictory zk proofs
```
Comment (seolaoh) [inline specs/challenge.md:49]:
```suggestion
- The deletion of a valid output due to two valid and contradictory ZK proofs
```

**Review by seolaoh — APPROVED** [review]

**Comment by 0xHansLee** [comment]
@kroma-network/l2-protocol ,
As you can see in this PR, there is a separate fix commit [5ac952c](https://github.com/kroma-network/kroma/pull/246/commits/5ac952c7a384e83bda0d2d7f3aec2e91292c90b9).
I think this is more helpful in reviewing the PR. How about you?
With this way of commits, the PR should be squashed and merged.

**Comment by seolaoh** [comment]
> @kroma-network/l2-protocol , As you can see in this PR, there is a separate fix commit [5ac952c](https://github.com/kroma-network/kroma/pull/246/commits/5ac952c7a384e83bda0d2d7f3aec2e91292c90b9). I think this is more helpful in reviewing the PR. How about you? With this way of commits, the PR should be squashed and merged.

Let's try it, then compare both ways

**Review by Pangssu — APPROVED** [review]

**Comment by 0xHansLee** [comment]
OK, then I will squash and merge this PR to dev branch

**Review by 0xHansLee — APPROVED** [review]
LGTM. Congrats your first PR in Kroma!

**Pull request merged**